src/interaction/index.ts

```diff
@@ -19,8 +19,22 @@ export function start(): void {
 }
 
 export function end(): void {
+    click.reset();
+    pointer.end();
     input.end();
     pointer.end();
     scroll.end();
     selection.end();
 }
+
+export function observe(root: Node): void {
+    scroll.observe(root);
+    // Only monitor following interactions if the root node is a document
+    // In case of shadow DOM, following events automatically bubble up to the parent document.
```

Comment:
Will these event inside the Shadow DOM be exposed here?

Reply:
As captured in the comment, shadow dom already bubbles up some of the events to parent DOM so we don't have to rewire them up.